flow/src/main/java/flow/History.java

```diff
@@ -207,7 +207,12 @@ private Builder(Collection<Entry> history) {
     }
 
     public Builder clear() {
-      history.clear();
+      // Clear by popping everything (rather than just calling history.clear()) to
+      // fill up entryMemory. Otherwise we drop view state on the floor.
+      while (!isEmpty()) {
```

Comment:
mmmm I don't know. I really really don't like that `pop()` keeps state. That means I don't have any way to remove entries **without** keeping state.

Why are we using clear() to rebuild a history where we expect to keep some state?